setup.py

```diff
@@ -14,6 +14,8 @@
 
 os.chdir(os.path.split(os.path.abspath(__file__))[0])
 
+# Try to get and update the version.
+# N.B. Similar code appears in docs/conf.py.
```

Comment:
In `conf.py` you're allowed to import the main modules, so maybe this could just be shoved in `hy/version_gen.py` or similar? (Not a blocking issue by any means, but still something to think about.)